execution/executor-types/src/in_memory_state_calculator.rs

```diff
@@ -134,21 +133,20 @@ impl InMemoryStateCalculator {
         HashMap<NibblePath, HashValue>,
         Option<HashValue>,
     )> {
-        let updated_state_keys = process_write_set(
-            Some(txn),
+        let updated_state_kvs = process_write_set(
+            &Some(txn),
             &mut self.state_cache,
             txn_output.write_set().clone(),
         )?;
-        self.updated_after_latest
-            .extend(updated_state_keys.into_iter());
+        self.updates_after_latest.extend(updated_state_kvs.clone());
         self.next_version += 1;
 
         if txn_output.is_reconfig() {
             self.checkpoint()
         } else {
             match txn {
                 Transaction::BlockMetadata(_) | Transaction::UserTransaction(_) => {
-                    Ok((HashMap::new(), HashMap::new(), None))
+                    Ok((updated_state_kvs, HashMap::new(), None))
                 }
                 Transaction::GenesisTransaction(_) | Transaction::StateCheckpoint => {
                     self.checkpoint()
```

Comment:
so a state change is reflected twice in the txns_to_commit, once at the version it actually happened, and again at the next state checkpoint??

Reply:
I also found this bug though correctness is not affected, I am gonna skip the state value commit at checkpoint in the fixup.